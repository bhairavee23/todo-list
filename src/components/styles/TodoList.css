.priority {
    background-color: white;
    border-color: #007bff;
    color: #007bff;
}