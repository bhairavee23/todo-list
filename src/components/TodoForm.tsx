import React, { useState } from 'react';
import { Todo } from './TodoContainer';

import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Button from 'react-bootstrap/Button';
import DropdownButton from 'react-bootstrap/DropdownButton';
import Dropdown from 'react-bootstrap/Dropdown';
import { nanoid } from 'nanoid';

export interface TodoFormProps {
    addTodo(todo: Todo): void;
}

export enum Priority {
    HIGH = 'HIGH',
    MEDIUM = 'MEDIUM',
    LOW = 'LOW'
}
export const TodoForm = (props: TodoFormProps) => {

    const initialTodo : Todo = {
        id: nanoid(),
        todoLabel : '',
        todoPriority : Priority.MEDIUM,
        completed: false
    }

    const [todo, setTodo] = useState<Todo>(initialTodo);

    const submitTodo = () => {
        props.addTodo(todo);
        console.log(todo);
        setTodo(initialTodo);
    }

    return (
        <div className="w-50 mx-auto">
            <InputGroup className="mb-3">
                <InputGroup.Prepend>
                    <DropdownButton
                        as={InputGroup.Append} 
                        variant="outline-primary"
                        title={todo.todoPriority}
                        id="priority-dropdown"
                    >
                        <Dropdown.Item onClick={() => setTodo({...todo, todoPriority: Priority.HIGH})}>High</Dropdown.Item>
                        <Dropdown.Item onClick={() => setTodo({...todo, todoPriority: Priority.MEDIUM})}>Medium</Dropdown.Item>
                        <Dropdown.Item onClick={() => setTodo({...todo, todoPriority: Priority.LOW})}>Low</Dropdown.Item>
                    </DropdownButton>
                </InputGroup.Prepend>
                <FormControl
                name="todoLabel"
                value={todo.todoLabel}
                placeholder="What are you doing today?"
                aria-label="What are you doing today?"
                aria-describedby=""
                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                    setTodo({...todo, todoLabel: e.target.value});
                }}
                onKeyUp={(e: React.KeyboardEvent<HTMLDivElement>) => {
                    if(e.key === 'Enter' && todo.todoLabel !== "") {
                        submitTodo();       
                    }
                }}
                />
                <InputGroup.Append>
                <Button variant="outline-success" onClick={submitTodo}>Add Todo</Button>
                </InputGroup.Append>
            </InputGroup>
        </div>
    )
}