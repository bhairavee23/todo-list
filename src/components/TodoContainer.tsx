import React, { useState } from 'react';
import { TodoForm } from './TodoForm';
import TodoList from './TodoList';

export interface Todo {
    id: string;
    todoLabel: string;
    todoPriority: string;
    completed: boolean;
}

export const TodoContainer = () => {
    const [todos, setTodos] = useState<Todo[]>([]);

    const addTodo = (todo: Todo) => {
        setTodos([...todos, todo]);
        console.log("this was the new todo : "+todo);
    }

    const deleteTodos = (completedTodos: Todo[]) => {
        console.log(completedTodos);
        console.log(todos.filter(todo => !completedTodos.includes(todo)));
        setTodos(todos.filter(todo => !completedTodos.includes(todo)));
    }
    
    return (
        <div className="p-5">
            <TodoForm addTodo={addTodo}/>
            <TodoList todos={todos} deleteTodos={deleteTodos}/>
        </div>
    )
}