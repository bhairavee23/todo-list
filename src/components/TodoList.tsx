import React, { useEffect, useState } from 'react';
import { Todo } from './TodoContainer'; 

import './styles/TodoList.css';
import { Button, FormControl, InputGroup } from 'react-bootstrap';

interface TodoListProps {
    todos: Todo[];
    deleteTodos(todos: Todo[]): void;
}

const TodoList = (props: TodoListProps) => {

    const {todos, deleteTodos} = props;
    const [currentTodos, setCurrentTodos] = useState<Todo[]>(todos);
    const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);

    useEffect(() => {
        setCurrentTodos(todos);
        setCompletedTodos([]);
    },[todos]);

    const deleteTodo = (t: Todo) => {
        if(completedTodos.length > 0) {
            deleteTodos(completedTodos);
        } else {
            t.completed = !t.completed;
            deleteTodos([t]);
        }
    }

    return (
        
        <div className={'w-50 mx-auto my-4'}>
            <div className="py-3"><h4>Todo List: </h4></div>
            {(currentTodos.length <= 0) ? 
                <div>All things on this list are done!</div> 
            : (
                currentTodos.map((t) => (
                    <div>
                        <InputGroup className="mb-3" key={t.id} id={t.id}>
                            <InputGroup.Prepend>
                                <InputGroup.Checkbox 
                                    aria-label="Checkbox for todo"
                                    key={t.todoLabel}
                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                                        if(e.target.checked) {
                                            t.completed  = true;
                                            setCompletedTodos([...completedTodos, t]);
                                        } else {
                                            t.completed  = false;
                                            setCompletedTodos(completedTodos.filter(todo => todo !== t));
                                        }
                                    }}
                                />
                            </InputGroup.Prepend>
                            <InputGroup.Prepend>
                                <InputGroup.Text className="priority" id="priority">{t.todoPriority}</InputGroup.Text>
                            </InputGroup.Prepend>
                            <FormControl
                            name="todoLabel"
                            value={t.todoLabel}
                            placeholder="What are you doing today?"
                            aria-label="What are you doing today?"
                            // onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                            //     setTodo({...todo, todoLabel: e.target.value});
                            // }}
                            // onKeyUp={(e: React.KeyboardEvent<HTMLDivElement>) => {
                            //     if(e.key === 'Enter' && todo.todoLabel !== "") {
                            //         submitTodo();       
                            //     }
                            // }}
                            />
                            <InputGroup.Append>
                            <Button variant="outline-danger" 
                                onClick={() => deleteTodo(t)}
                            >
                                Completed
                            </Button>
                            </InputGroup.Append>
                        </InputGroup>
                    </div>
                ))
            )
            }
        </div>
    )
};

export default TodoList;   