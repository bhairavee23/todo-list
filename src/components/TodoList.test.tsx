import React from 'react';
import renderer from 'react-test-renderer';
import { Todo } from './TodoContainer';
import { Priority } from './TodoForm';
import TodoList from './TodoList';

describe('Test TodoList', () => {

    it("should match snapshot", ()=> {
        const tree = renderer.create(
            <TodoList
                todos={[]}
                deleteTodos={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot(); 
    });     
    
    it("should have two todos", ()=> {
        const todoData: Todo[] = [{
            id: "1",
            todoLabel:"todo1",
            todoPriority: Priority.MEDIUM,
            completed: false
        }, {
            id: '2',
            todoLabel:"todo1",
            todoPriority: Priority.LOW,
            completed: false
        }]
        const tree = renderer.create(
            <TodoList
                todos={todoData}
                deleteTodos={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot(); 
    });    
});