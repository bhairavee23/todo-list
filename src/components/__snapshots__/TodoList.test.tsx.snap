// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoList should have two todos 1`] = `
<div
  className="w-50 mx-auto my-4"
>
  <div
    className="py-3"
  >
    <h4>
      Todo List: 
    </h4>
  </div>
  <div>
    <div
      className="mb-3 input-group"
      id="1"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <input
            aria-label="Checkbox for todo"
            onChange={[Function]}
            type="checkbox"
          />
        </span>
      </div>
      <div
        className="input-group-prepend"
      >
        <span
          className="priority input-group-text"
          id="priority"
        >
          MEDIUM
        </span>
      </div>
      <input
        aria-label="todo1"
        className="form-control"
        name="todoLabel"
        value="todo1"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-outline-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Completed
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="mb-3 input-group"
      id="2"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <input
            aria-label="Checkbox for todo"
            onChange={[Function]}
            type="checkbox"
          />
        </span>
      </div>
      <div
        className="input-group-prepend"
      >
        <span
          className="priority input-group-text"
          id="priority"
        >
          LOW
        </span>
      </div>
      <input
        aria-label="todo1"
        className="form-control"
        name="todoLabel"
        value="todo1"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-outline-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Completed
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test TodoList should match snapshot 1`] = `
<div
  className="w-50 mx-auto my-4"
>
  <div
    className="py-3"
  >
    <h4>
      Todo List: 
    </h4>
  </div>
  <div>
    All things on this list are done!
  </div>
</div>
`;
